.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+name: Build and test gem
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  cross-compile:
+    strategy:
+      matrix:
+        platform:
+          - "x64-mingw32"
+          - "x86-mingw32"
+          - "x64-mingw-ucrt"
+    name: Cross-compile gem
+    runs-on: ubuntu-latest
+    container:
+      image: "ghcr.io/rake-compiler/rake-compiler-dock-image:1.4.0-mri-${{ matrix.platform }}"
+    steps:
+      - uses: actions/checkout@v4
+      - run: git config --global --add safe.directory /__w/tiny_tds/tiny_tds # shrug
+      - name: Install gems
+        shell: bash
+        run: bundle install
+      - name: Write used versions into file
```

Comment:
My fake undiagnosed OCD demands more newlines for this job, same as you have the others 😆